### ISSUE TITLE
read!(io::IO, ::Type{SA})

### ISSUE THREAD

**Issue by tpapp** [JuliaArrays/StaticArrays.jl#505]
Why is [this](https://github.com/JuliaArrays/StaticArrays.jl/blob/86097997bd846f0b9f4331cd55ba8a233375173b/src/io.jl#L2) a method of `read!`, instead of `read`? It does not modify any of its arguments.

**Comment by andyferris** [comment]
Good point. We should have a `read!` method for `MArray`, `SizedArray`, etc - but yes, `read` for immutable arrays.

**Comment by c42f** [comment]
Yes the `read()` API makes perfect sense for `StaticArray`s ­— removing `read` was a small part of #451 which should be rolled back.
